src/XML/src/FMI3/fmi3_xml_type_impl.h

```diff
@@ -116,16 +132,19 @@ typedef enum {
  * 'nextLayer'. An attempt to show the ordering is made in the 'diagrams.drawio' file.
  *
  * More info can also be found at the top of the file where this type is described.
+ *
+ * Note that all objects of this type are appended to a list owned by the type definitions object.
  */
 typedef struct fmi3_xml_variable_type_base_t fmi3_xml_variable_type_base_t;
 struct fmi3_xml_variable_type_base_t {
-    fmi3_xml_variable_type_base_t* nextLayer;   /* the next layer in the type aggregate */
-    fmi3_xml_type_struct_kind_enu_t structKind; /* the actual (sub) type */
-    fmi3_base_type_enu_t baseType;              /* the FMI base type */
-    char isRelativeQuantity;                    /* relativeQuantity flag (only used in fmi3_xml_real_type_props_t) */
-    char isUnbounded;                           /* unbounded flag        (only used in fmi3_xml_real_type_props_t) */
-
-    fmi3_xml_variable_type_base_t* next;        /* should only be used for deallocation */
+    fmi3_xml_variable_type_base_t* nextLayer;   /* The next layer in the type aggregate */ // TODO: Rename to next
+    fmi3_xml_type_struct_kind_enu_t structKind; /* The actual (sub) type */
+    fmi3_base_type_enu_t baseType;              /* The FMI base type */
+    char isRelativeQuantity;                    /* RelativeQuantity flag (only used in fmi3_xml_real_type_props_t) */
+    char isUnbounded;                           /* Unbounded flag        (only used in fmi3_xml_real_type_props_t) */
+
+    // TODO: Remove (create a new jm_vector on typeDefinitions that works the same)
+    fmi3_xml_variable_type_base_t* next;        /* For deallocation: the next node in the deallocation list */
```

Comment:
Would also cause array ticket conflicts. And make this ticket harder to review than it already is.